Tests/SwiftGRPCNIOTests/GRPCChannelHandlerTests.swift

```diff
@@ -0,0 +1,195 @@
+import Foundation
+import XCTest
+import NIO
+import NIOHTTP1
+@testable import SwiftGRPCNIO
+
+func gRPCMessage(channel: EmbeddedChannel, compression: Bool = false, message: Data? = nil) -> ByteBuffer {
+  let messageLength = message?.count ?? 0
+  var buffer = channel.allocator.buffer(capacity: 5 + messageLength)
+  buffer.write(integer: Int8(compression ? 1 : 0))
+  buffer.write(integer: UInt32(messageLength))
+  if let bytes = message {
+    buffer.write(bytes: bytes)
+  }
+  return buffer
+}
+
+class GRPCChannelHandlerTests: GRPCChannelHandlerResponseCapturingTestCase {
+  func testUnimplementedMethodReturnsUnimplementedStatus() throws {
+    let responses = try waitForGRPCChannelHandlerResponses(count: 1) { channel in
+      let requestHead = HTTPRequestHead(version: .init(major: 2, minor: 0), method: .POST, uri: "unimplementedMethodName")
+      try channel.writeInbound(RawGRPCServerRequestPart.head(requestHead))
+    }
+
+    let expectedError = GRPCError.unimplementedMethod("unimplementedMethodName")
+    XCTAssertEqual(expectedError, errorCollector.errors.first as? GRPCError)
```

Comment:
`XCTAssertEqual([expectedError], errorCollector.errors as? [GRPCError])`, to ensure that we aren't logging extra errors that we don't expect?

Reply:
Good idea!